### PR TITLE
Refactored to support device info function

### PR DESCRIPTION
**Pull request by milanperera**

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java [hunk 1]
```diff
@@ -378,9 +378,10 @@ public Device getDevice(DeviceIdentifier deviceId) throws DeviceManagementExcept
     }
 
     @Override
-    public boolean updateDeviceInfo(Device device) throws DeviceManagementException {
-        DeviceManager dms = this.getPluginRepository().getDeviceManagementProvider(device.getType());
-        return dms.updateDeviceInfo(device);
+    public boolean updateDeviceInfo(DeviceIdentifier deviceIdentifier, Device device) throws DeviceManagementException {
```
Comment (DilanUA) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java:381]: updateDeviceInfo method does not sound like returning a boolean?

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementServiceImpl.java [hunk 2]
```diff
@@ -98,8 +98,10 @@ public Device getCoreDevice(DeviceIdentifier deviceId) throws DeviceManagementEx
     }
 
     @Override
-    public boolean updateDeviceInfo(Device device) throws DeviceManagementException {
-        return DeviceManagementDataHolder.getInstance().getDeviceManagementProvider().updateDeviceInfo(device);
+    public boolean updateDeviceInfo(DeviceIdentifier deviceIdentifier, Device device) throws DeviceManagementException {
```
Comment (DilanUA) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementServiceImpl.java:101]: updateDeviceInfo method does not sound like returning a boolean?

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java (hunk 1 above)
Comment (milanperera) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java:381]: In AndroidDeviceDAO, it is implemented to return true when a device successfully add, remove or update. Hence it is being used here as well to acknowledge.

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementServiceImpl.java (hunk 2 above)
Comment (milanperera) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementServiceImpl.java:101]: In AndroidDeviceDAO, it is implemented to return true when a device successfully add, remove or update. Hence it is being used here as well to acknowledge.

**Pull request merged**